src/coreclr/vm/jithelpers.cpp

```diff
@@ -2100,31 +2045,6 @@ HCIMPLEND
 //
 //========================================================================
 
-/*********************************************************************/
-// Called by the JIT whenever a cee_break instruction should be executed.
-//
-HCIMPL0(void, JIT_UserBreakpoint)
-{
-    FCALL_CONTRACT;
-
-    HELPER_METHOD_FRAME_BEGIN_NOPOLL();    // Set up a frame
-
-#ifdef DEBUGGING_SUPPORTED
-    FrameWithCookie<DebuggerExitFrame> __def;
```

Comment:
Does it matter that we won't have this Frame type on the Frame chain now?

Reply:
Looked at the comments and it looks like it should be fine.

Can we update the comment on DebuggerExitFrame to better reflect remaining usages?

Reply:
I've no idea whether it is needed or not. I tagged the diagnostics team to help review this. If it is needed, then I will convert this to a QCall rather than C#.

Reply:
> Can we update the comment on DebuggerExitFrame to better reflect remaining usages?

+1

I have noticed that `DebuggerSecurityCodeMarkFrame` is dead code. You can delete it while you are on it.

Reply:
> I've no idea whether it is needed or not. I tagged the diagnostics team to help review this. If it is needed, then I will convert this to a QCall rather than C#.

I doubt the FrameWithCookie needs to be preserved but to increase confidence I'd suggest creating an app that executes this helper, run it under VS debugger and confirm:
- When the debugger stops at the breakpoint, the callstack shown in VS looks the same as it used to.
- If you manually set a breakpoint just prior to the user-break and step-in/step-over, that behavior is the same as before.

Reply:
@noahfalk Thanks. I was able to confirm that the current behavior is identical to the previous FCall impl.